src/main/java/fi/helsinki/cs/tmc/core/communication/serialization/AdaptiveExerciseParser.java

```diff
@@ -1,38 +1,45 @@
 package fi.helsinki.cs.tmc.core.communication.serialization;
 
-import fi.helsinki.cs.tmc.core.domain.Exercise;
+import fi.helsinki.cs.tmc.core.domain.AdaptiveExercise;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 
-import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.net.URI;
 
 public class AdaptiveExerciseParser {
-    
+
     private static final Logger logger = LoggerFactory.getLogger(AdaptiveExerciseParser.class);
 
-    public Exercise parseFromJson(String json) {
+    public AdaptiveExercise parseFromJson(String json) {
         if (json == null) {
             throw new NullPointerException("Json string is null");
         }
         if (json.trim().isEmpty()) {
             throw new IllegalArgumentException("Empty input");
         }
+        if (json.contains("\"available\":\"false\"")) {
```

Comment:
Why are you doing this here? The check could be accomplished by using the `getAvailable` method after the json has been parsed.